source/common/http/http1/codec_impl.cc

```diff
@@ -788,8 +788,9 @@ Envoy::StatusOr<int> ConnectionImpl::onHeadersCompleteBase() {
 
   header_parsing_state_ = HeaderParsingState::Done;
 
-  // Returning 2 informs http_parser to not expect a body or further data on this connection.
-  return handling_upgrade_ ? 2 : statusor.value();
+  // Returning HttpParserCode::NoBodyData informs http_parser to not expect a body or further data
+  // on this connection.
+  return handling_upgrade_ ? enumToSignedInt(HttpParserCode::NoBodyData) : statusor.value();
```

Comment:
Thanks for doing this, this is definitely a good place to have readable values.

I wonder if https://github.com/envoyproxy/envoy/blob/b11b3c11f5d64230fb5f419e0999c60866e9e30c/source/common/http/http1/codec_impl.cc#L410 should also be changed away from unsigned int?

Reply:
I also found that counter intuitive that `enumToInt` returned an unsigned int.

The enum in question is an enum class: https://github.com/envoyproxy/envoy/blob/1d1b708c7bf6efa02c41d9ce22cbf1e4a1aeec2c/source/common/http/http1/codec_impl.h#L243